### PR TITLE
fix: get schema name from profile or override from CLI

### LINKED ISSUE #98: [bug] schema in run_test should not be read from flags
In #85 we merged code that obtains the schema of the model we want to documents from the CLI. While we offer this argument on the CLI it's discouraged.

We should be reading the schema from the dbt profile. We require a `target_profile` argument to be given, if none is given or user passes it in the CLI we should override the field in the dbt profile object with the content of what was passed into the CLI but the test backend should not be reading directly from flags and rather read from the dbt profile which should be responsible for overriding things appropriately given relevant CLI inputs.

### PR DESCRIPTION
**Pull request by bastienboutonnet**
# Description

The model's schema was read from the CLI `--schema` argument only. 
This is actually discouraged in our documentation since we prefer to limit surprises and read the target schema directly from the dbt profile.

As of this PR we now read the schema from the dbt profile and **if and only if the argument is provided to the CLI** we surrender to the CLI argument because it is "closer" to the user's scope

# How was the change tested
Unit tests were updated and passing locally.

# Issue Information
Closes #98

# Checklist

(Ideally, all boxes are checked by the time we merged the PR, if you don't know how to do any of these don't hesitate to say so in the PR and we'll help you out.)

- [x] I formatted my PR name according to [CONTRIBUTING.md](CONTRIBUTING.md)
- [x] I added "Closes #<issue_number>" in the "Issue Information" section (if no issue, feel free to tick thick the box anyway).

### CODE REVIEW

**Review by virvirlopez — APPROVED** [review]
LGTM

File: dbt_sugar/core/task/doc.py [hunk 1]
```diff
@@ -35,6 +35,8 @@ def __init__(self, flags: FlagParser, dbt_profile: DbtProfile, config: DbtSugarC
         self._dbt_profile = dbt_profile
         self._sugar_config = config
 
+    # FIXME: Ask Virginia why this is here, I don't think it's needed since we already read the
```
Comment (bastienboutonnet) [inline dbt_sugar/core/task/doc.py:38]: @virvirlopez Was wondering why this is here since we already read `DbtProfile` in main. Can you get back to me?

Comment (virvirlopez) [reply dbt_sugar/core/task/doc.py:38]: to connect to the DB, to run the queries.

Comment (bastienboutonnet) [reply dbt_sugar/core/task/doc.py:38]: Yeah but it's already available for you in main so I don't think we need to re-create it here do we?

**Review by virvirlopez — APPROVED** [review]

**Pull request merged**